### ISSUE TITLE
Fonts: use a `ConcurrentDictionary` for caching

### ISSUE THREAD

**Issue by Hawkwind250** [ScottPlot/ScottPlot#3940]
**Issue**:  InvalidOperationException raised when calling GetImageBytes on different plot instances in concurrent environment
**ExceptionMessage**:  Operations that change non-concurrent collections must have exclusive access. A concurrent update was performed on this collection and corrupted its state. The collection's state is no longer correct.
**Stack Trace**:
```
   at System.ThrowHelper.ThrowInvalidOperationException_ConcurrentOperationsNotSupported()
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at ScottPlot.Fonts.GetTypeface(String fontName, Boolean bold, Boolean italic)
   at ScottPlot.Label.ApplyTextPaint(SKPaint paint)
   at ScottPlot.Label.ApplyToPaint(SKPaint paint)
   at ScottPlot.AxisPanels.XAxisBase.RegenerateTicks(PixelLength size)
   at ScottPlot.LayoutEngines.Automatic.<>c__DisplayClass1_0.<GetLayout>b__0(IXAxis x)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at ScottPlot.LayoutEngines.Automatic.GetLayout(PixelRect figureRect, Plot plot)
   at ScottPlot.RenderPack.CalculateLayout()
   at ScottPlot.Rendering.RenderActions.CalculateLayout.Render(RenderPack rp)
   at ScottPlot.Rendering.RenderManager.Render(SKCanvas canvas, PixelRect rect)
   at ScottPlot.Plot.Render(SKCanvas canvas, PixelRect rect)
   at ScottPlot.Plot.Render(SKCanvas canvas, Int32 width, Int32 height)
   at ScottPlot.Plot.GetImage(Int32 width, Int32 height)
   at ScottPlot.Plot.GetImageBytes(Int32 width, Int32 height, ImageFormat format)
```
Also following exception might happen, depending on your luck: `An item with the same key has already been added. Key: (Segoe UI, False, False)` leading to the same source code:

https://github.com/ScottPlot/ScottPlot/blob/f0279f28c203218415f6f7b0a51606dab76bc36d/src/ScottPlot5/ScottPlot5/Fonts.cs#L15

**Suggestion**: Change `Dictionary<(string, bool, bool), SKTypeface>` type to `ConcurrentDictionary<(string, bool, bool), SKTypeface>`

Code to reproduce an issue (in form of minimal console app):

```csharp
const int tasksAmount = 10;
const int samplesAmount = 10;
const int sampleMaxValue = 1000;

var random = new Random(100500); // constant seed for reproducible results

var tasks = new Task[tasksAmount];
for (var i = 0; i < tasksAmount; i++)
{
    var samples = GenerateSamples(samplesAmount);
    tasks[i] = Task.Run(() => GetImageBytes(samples, 100, 20));
}

await Task.WhenAll(tasks);

IReadOnlyList<int> GenerateSamples(int samplesAmount)
{
    var result = new int[samplesAmount];
    
    for (var i = 0; i < samplesAmount; i++)
    {
        result[i] = random.Next(sampleMaxValue);
    }
    
    return result;
}

byte[] GetImageBytes(IReadOnlyList<int> samples, int width, int height)
{
    var max = samples.Max();

    var plot = new Plot
    {
        DataBackground = new BackgroundStyle { Color = Colors.Transparent }
    };

    var coordinates = new Coordinates[samples.Count];
    for (var i = 0; i < samples.Count; i++)
    {
        var sample = samples[i];
        coordinates[i] = new Coordinates(i, 100.0 * sample / max);
    }

    var scatter = plot.Add.ScatterLine(coordinates);
    scatter.LineWidth = 1;
    scatter.PathStrategy = new CubicSpline();

    plot.Axes.Left.IsVisible = false;
    plot.Axes.Right.IsVisible = false;
    plot.Axes.Bottom.IsVisible = false;
    plot.Axes.Top.IsVisible = false;

    plot.Grid.IsVisible = false;
    return plot.GetImageBytes(width, height, ImageFormat.Png);
}
```

**Comment by swharden** [comment]
Thanks @Hawkwind250 for reporting this issue and identifying a fix! The sample code was really helpful too. I fixed this in 11d03e6 and it will be in the next release 👍

**Comment by Hawkwind250** [comment]
> Thanks @Hawkwind250 for reporting this issue and identifying a fix! The sample code was really helpful too. I fixed this in [11d03e6](https://github.com/ScottPlot/ScottPlot/commit/11d03e62d3f2d1cfabc155b4b985360fd4c55897) and it will be in the next release 👍

Wow, this was really fast, thanks! 👍
